docs/coding-practices.md

```diff
@@ -0,0 +1,105 @@
+
+## SOLUTION:
+
+### Linting
+
+First of all, there is a linting standard that we'd like to respect. We'll have one for both parts : backend and frontend
+
+#### Back-End
+
+In the back-end we'll be using golang-ci-lint that has a respected standard. We'll be using the default linting that it proposes but this may be subject to be modified in the futur.
+
+- [errcheck](https://github.com/kisielk/errcheck) : Errcheck is a program for checking for unchecked errors in Go code. These unchecked errors can be critical bugs in some cases.
+- [gosimple ⚙️](https://github.com/dominikh/go-tools/tree/master/simple) : Linter for Go source code that specializes in simplifying code.
+- [govet ⚙️](https://pkg.go.dev/cmd/vet) : Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string.
+- [ineffassign](https://golangci-lint.run/usage/linters/#:~:text=v1.0.0-,ineffassign,-Detects%20when%20assignments) : Detects when assignments to existing variables are not used.
+- [staticcheck ⚙️](https://golangci-lint.run/usage/linters/#:~:text=v1.0.0-,staticcheck,-%E2%9A%99%EF%B8%8F) : It's a set of rules from staticcheck. It's not the same thing as the staticcheck binary. The author of staticcheck doesn't support or approve the use of staticcheck as a library inside golangci-lint.
+- [unused ⚙️](https://github.com/dominikh/go-tools/tree/master/unused) : Checks Go code for unused constants, variables, functions and types.
+
+### Case type & practices
+
+#### Global
+
+- Follow `camelCase` coding standard.
+- Always re-use, upgrade, or improve the existing code. The core can be modified but make sure to refactor affected code and always think how the code can be scalable.
+- Always start the function name with an action
+
+#### Back-End
+
+- If needed for constants, declare them like and where it's used :
+
+```go
+const (
+	stagingEnv    string = "staging"
+	productionEnv string = "production"
+)
+```
+
+- When there are more than 3 data types to return. Make a struct to return your data, you should do something like this :
+
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+)
+
+type Song struct {
+	Name         string
+	Artist       string
+	Album        string
+	Genre        string
+	AmountPlayed int
+}
+
+func GetSong(name string) (Song, error) {
+	return Song{
+		Name: "Pink + White",
+		Artist: "Frank Ocean",
+		Album: "Blond",
+		Genre: "R&B",
+		AmountPlayed: 1 000 000 000,
+	}, nil
+}
+
+func main() {
+	song, err := getSong()
+	if err != nil {
+		return errors.Wrapf(err, "Main, Unable to getSong")
+	}
+	fmt.Println(song.Name)
+
+}
+```
+
+### Code reusability
+
+#### Back-End
+
+**When coding :**
+
+- Make public functions in Go only when needed. (Start of function should be lowercased to make it private)
+
+### Error Handling
+
+#### Back-End
+
+- Always add context to errors.
```

Comment:
Can you provide more explainations please?

Reply:
Explanation is in the errors section